### PR TITLE
Analytics for litmus portal.

### PR DESCRIPTION
**Pull request by ishangupta-ds**
## Proposed changes 

Adds analytics section and fixes for the same to litmus portal. 

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

## Dependency
- None.

## Special notes for your reviewer:
Analytics PR #3

### CODE REVIEW

**Comment by ishangupta-ds** [comment]
@amityt @rajdas98 @arkajyotiMukherjee @S-ayanide @Jonsy13 PTAL. cc: @ajeshbaby

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/CustomStatus/styles.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    textAlign: 'center',
+    borderRadius: 3,
+    paddingTop: theme.spacing(0.375),
+    paddingBottom: theme.spacing(0.375),
+    color: theme.palette.primary.dark,
+    backgroundColor: theme.palette.customColors.menuOption.active,
+    verticalAlign: 'middle',
+    display: 'inline-flex',
+  },
+  failed: {
```
Comment (Jonsy13) [inline litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/CustomStatus/styles.ts:15]: CSS for passed and failed are same, only backgroundColor is different, It can avoided by defining a separate class for common parameters and just concatenating to it the different one.
You can use the  `${} ${}`  sysntax. cc: @arkajyotiMukherjee

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/CustomStatus/styles.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    backgroundColor: theme.palette.secondary.contrastText,
+    width: '0.9375rem',
+    height: '0.9375rem',
+  },
+  cancelIcon: {
+    color: theme.palette.error.dark,
+    width: '0.9375rem',
+    height: '0.9375rem',
+  },
+  checkIcon: {
```
Comment (Jonsy13) [inline litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/CustomStatus/styles.ts:43]: Same for these  icons.

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/TableToolbar.tsx [hunk 3]
```diff
@@ -0,0 +1,251 @@
+import {
+  Typography,
+  InputBase,
+  InputAdornment,
+  FormControl,
+  InputLabel,
+  MenuItem,
+  Select,
+  Button,
```
Comment (Jonsy13) [inline litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/TableToolbar.tsx:9]: If it is special use case for not using custom button then ignore otherwise PTAL @ishangupta-ds

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/TableToolbar.tsx [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+          <MenuItem value="All">All</MenuItem>
+          {clusters.map((cluster: string) => (
+            <MenuItem value={cluster}>{cluster}</MenuItem>
+          ))}
+        </Select>
+      </FormControl>
+
+      {compare === false || comparisonState === false ? (
+        <Button
+          variant="outlined"
```
Comment (Jonsy13) [inline litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/TableToolbar.tsx:215]: I think custom one can be used here?

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/TableHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+                  })
+                }
+              >
+                <ExpandMoreTwoToneIcon className={classes.markerIconDown} />
+              </IconButton>
+            </div>
+          </div>
+        </TableCell>
+        <TableCell>
+          <Button
```
Comment (Jonsy13) [inline litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/TableHeader.tsx:186]: Custom one?

**Review by Jonsy13 — CHANGES_REQUESTED** [review]

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/CustomStatus/Status.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import React, { useEffect } from 'react';
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/CustomStatus/Status.tsx:1]: avoid naming components as Custom

Comment (arkajyotiMukherjee) [reply litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/CustomStatus/Status.tsx:1]: change it to a more apt name

File: litmus-portal/frontend/src/components/DateRangeSelector/index.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
+/* eslint-disable react/no-array-index-key */
```
Comment (amityt) [inline litmus-portal/frontend/src/components/DateRangeSelector/index.tsx:1]: Why is this required? I've used the same date range picker earlier, I don't think we need this.

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/CustomStatus/Status.tsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+const CustomStatus: React.FC<StatusProps> = ({ status }) => {
+  const classes = useStyles();
+  const [label, setLabel] = React.useState(' ');
+
+  useEffect(() => {
+    if (status === 'Passed') {
+      return setLabel(classes.passed);
+    }
+    return setLabel(classes.failed);
+  }, [status, classes.failed, classes.passed]);
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/CustomStatus/Status.tsx:20]: classes can be removed as a dependency

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/TableToolbar.tsx (hunk 3 above)
Comment (arkajyotiMukherjee) [reply litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/TableToolbar.tsx:9]: same

**Review by Saranya-jena — DISMISSED** [review]

File: litmus-portal/frontend/src/components/DateRangeSelector/index.tsx (hunk 7 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/DateRangeSelector/index.tsx:1]: It was throwing some error in eslint so i ignored this rule .. @amityt it is needed for eslint checks to pass..

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/styles.ts [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+
+  search: {
+    marginRight: 'auto',
+    marginLeft: theme.spacing(6.25),
+    borderBottom: '1px solid rgba(0, 0, 0, 0.1)',
+    width: '17.5rem',
+  },
+
+  calIcon: {
+    color: 'rgba(0, 0, 0, 0.4)',
```
[the remark this answers is not in the document]
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/styles.ts:61]: No we can't @Saranya-jena custom colours are inaccessible and throwing error in this component, after this release we will try to debug this, for a few components it happening not for all, i had earlier used custom colours but had to change because of the crash cc: @arkajyotiMukherjee

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/styles.ts [hunk 10]
```diff
@@ -0,0 +1,290 @@
...
+    width: '0.625rem',
+    height: '0.625rem',
+  },
+}));
+
+export const customTheme = createMuiTheme({
+  palette: {
+    secondary: {
+      light: '#2CCA8F',
+      main: '#2CCA8F',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/styles.ts:272]: what is theme data doing outside theme file

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/styles.ts [hunk 11]
```diff
@@ -0,0 +1,290 @@
...
+      contrastText: '#000000',
+    },
+  },
+});
+
+export const customThemeCompare = createMuiTheme({
+  palette: {
+    secondary: {
+      light: '#FFFFFF',
+      main: '#FFFFFF',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/styles.ts:283]: same here

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/styles.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (Saranya-jena) [reply litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/styles.ts:61]: did you try like this? `border: `1px solid ${theme.palette.grey[500]}`,` it should work

[the remark this answers is not in the document]
Comment (arkajyotiMukherjee) [reply litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/styles.ts:61]: try this @ishangupta-ds

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/index.tsx [hunk 12]
```diff
@@ -0,0 +1,396 @@
...
+          (points.reduce((a, b) => a + b, 0) /
+            weights.reduce((a, b) => a + b, 0)) *
+          100
+        ).toFixed(2)
+      )
+    );
+    getTests(processedWorkflowRunDetails);
+    getTestResults(processedWorkflowRunDetails);
+    setReload(true);
+  }, [data, loading, workflowID, reload]);
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/index.tsx:200]:
```suggestion
  }, [data, workflowID, reload]);
```

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/index.tsx [hunk 13]
```diff
@@ -0,0 +1,396 @@
...
+
+  useEffect(() => {
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/index.tsx:202]: filter is a state variable it'll automatically recalculate this. no need for useEffect here

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/TableToolbar.tsx [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
+import useStyles from './styles';
+import DateRangeSelector from '../../../../components/DateRangeSelector';
+
+interface TestCallBackType {
+  (clusterName: string): void;
+}
+
+interface TestResultCallBackType {
+  (testResult: string): void;
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/TableToolbar.tsx:24]: improve this callback situation wherever it is used

**Review by arkajyotiMukherjee — DISMISSED** [review]

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/styles.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/styles.ts:61]: There will be a minor difference in the colour then @Saranya-jena i have picked colours from the figma so.. not sure if making changes to it will be good just for the sake of custom colours, i guess debugging the custom colours issue will be better and more appropriate...  @arkajyotiMukherjee and i have decided to debug that after release we will fix this later and use custom colours from theme.

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/styles.ts (hunk 10 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/styles.ts:272]: It is required @arkajyotiMukherjee for the comparison table as it switches colour for the table component if i don't use this i will have to add css to each sub component of the table.

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/styles.ts (hunk 11 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowComparisonTable/styles.ts:283]: It is required @arkajyotiMukherjee do i move it to theme file?

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/index.tsx (hunk 13 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/index.tsx:202]: I am updating the selection options dynamically upon changing he filter @arkajyotiMukherjee , if i don't use useEffect() how should i call the functions to update the drop down lists for test and tet results?

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/TableToolbar.tsx (hunk 14 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/TableToolbar.tsx:24]: Actually its more a state passing function than callback, have named it a callback as it functions similarly.. @arkajyotiMukherjee

**Comment by ishangupta-ds** [comment]
@arkajyotiMukherjee PTAL. cc: @rajdas98

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/index.tsx (hunk 13 above)
Comment (arkajyotiMukherjee) [reply litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/index.tsx:199]: this is basic concept of state in react...we have done the same in browse workflow
filter is a state var. hence when it changes react will rerender the component where it is used...no need of useEffect

**Review by arkajyotiMukherjee — CHANGES_REQUESTED** [review]

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/index.tsx (hunk 13 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/index.tsx:199]: That is good but I am calling 3 functions PTAL, how do I call them without use effect?

**Comment by ishangupta-ds** [comment]
changes made PTAL @arkajyotiMukherjee @rajdas98

**Review by arkajyotiMukherjee — APPROVED** [review]

**Review by amityt — APPROVED** [review]

**Review by amityt — DISMISSED** [review]

**Review by arkajyotiMukherjee — DISMISSED** [review]
